src/lib/assets/assets.test.ts

```diff
@@ -240,44 +267,63 @@ describe("Assets", () => {
   });
 
   it("should call generateHelmChart which should call createDirectoryIfNotExists twice for templates and charts", async () => {
-    const webhookGeneratorFunction = jest
-      .fn<() => Promise<V1MutatingWebhookConfiguration | V1ValidatingWebhookConfiguration | null>>()
-      .mockResolvedValue(new kind.MutatingWebhookConfiguration());
-    await assets.generateHelmChart(webhookGeneratorFunction, "/tmp");
+    const webhookGeneratorFunction = createMockWebhookGenerator();
+    const getWatcherFunction = createMockWatcher();
+    const getModuleSecretFunction = createMockModuleSecret();
+
+    await assets.generateHelmChart(
+      webhookGeneratorFunction,
+      getWatcherFunction,
+      getModuleSecretFunction,
+      "/tmp",
+    );
```

Comment:
This usage pattern _could_ get unwieldy as the amount of injected functions grows, but we have a linter rule to caution against more than 4 parameters to any function.